From 1509b223c391c7ed9a5877e73c719f3c4aadff4e Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Thu, 22 Aug 2024 12:14:22 +0300
Subject: [PATCH] [SW-197979] gaudi3: added a support of image validation at erom

This patch add a support of image validation at erom, before loading
the new firmware.

libfdt library expects public key dtb address must be aligned to
8byte. Otherwise, fdt_check_header(pub_key_fit) will fail and not
load any new firmware. So, aligned that fit_sec variable to 8byte.

Added preboot_img_validation_is_enabled() for gaudi3 erom.

Change-Id: Iebaaa1735a272cdc7ce421a7a7e2e8a5b2990588
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index bae5d65..be1cb1f 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1588,3 +1588,9 @@
 
 	return idx;
 }
+
+/* TODO: move it to emb-common-mods */
+bool preboot_is_image_sig_supported(void)
+{
+	return true;
+}
diff --git a/lib/common/security.c b/lib/common/security.c
index e0bcff5..4029ed5 100644
--- a/lib/common/security.c
+++ b/lib/common/security.c
@@ -13,7 +13,8 @@
 #include <string.h>
 #include <uart.h>
 
-static struct hl_fit_sec_desc fit_sec;
+/* libfdt is execting pub_key_fit dtb address must be 8 byte aligned. */
+static struct hl_fit_sec_desc fit_sec __aligned(8);
 static u8 signature[FIT_SIGNATURE_SIZE];
 
 
@@ -38,7 +39,7 @@
 	const void *fit_sec_off;
 	const void *pub_key_fit;
 
-	if (!preboot_security_is_enabled())
+	if (!preboot_img_validation_is_enabled())
 		return 0;
 
 	/* hl fit files are padded so the size is 4 bytes aligned */
